src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionMidEventHandler.java

```diff
@@ -60,4 +93,60 @@ private boolean isDecisionNoticeDateOfDecisionInvalid(SscsCaseData sscsCaseData)
         return false;
     }
 
+    @SuppressWarnings("squid:S1172")
+    @Override
+    protected DirectionOrDecisionIssuedTemplateBody createPayload(SscsCaseData caseData, String documentTypeLabel, LocalDate dateAdded, boolean isScottish,
+        String userAuthorisation) {
+        DirectionOrDecisionIssuedTemplateBody formPayload = super.createPayload(caseData, documentTypeLabel, dateAdded, isScottish, userAuthorisation);
+
+        DirectionOrDecisionIssuedTemplateBodyBuilder builder = formPayload.toBuilder();
+
+        builder.heldBefore(buildHeldBefore(caseData, userAuthorisation));
+
+        if (caseData.getHearings() != null && !caseData.getHearings().isEmpty()) {
```

Comment:
Minor point but you could use `CollectionUtils.isNotEmpty(caseData.getHearings())` here to avoid having 2 statements

Reply:
Much nicer - thanks!